--- set-java-home.zsh	2025-07-17 14:31:52.112691037 +0200
+++ set-java-home.zsh.good	2025-08-23 21:53:31.608626260 +0200
@@ -1,12 +1,19 @@
+###### fix for https://github.com/halcyon/asdf-java/issues/244
 asdf_update_java_home() {
   local java_path
   java_path="$(asdf which java)"
   if [[ -n "${java_path}" ]]; then
     export JAVA_HOME
-    JAVA_HOME="$(dirname "$(dirname "${java_path:A}")")"
+    if [[ "$java_path" == "/usr/bin/java" ]]; then
+      JAVA_HOME="$(/usr/libexec/java_home)"
+    else
+      JAVA_HOME="$(dirname "$(dirname "${java_path:A}")")"
+    fi
+    export JAVA_HOME=${JAVA_HOME}
     export JDK_HOME=${JAVA_HOME}
   fi
 }
 
 autoload -U add-zsh-hook
 add-zsh-hook precmd asdf_update_java_home
+###### p_a_t_c_h_e_d
\ No newline at end of file
