--- _asdf	2025-08-23 07:18:14
+++ _asdf.good	2025-08-23 07:42:53
@@ -1,6 +1,6 @@
 #compdef asdf
 #description tool to manage versions of multiple runtimes
-
+#P_A_T_C_H_E_D
 # Initialize local variables for ZSH completion context
 local state subcmd
 
@@ -310,12 +310,13 @@
           _asdf__installed_plugins
         else
           # Complete with available versions for the plugin
-          local versions
-          if versions=$(asdf list all "${words[3]}" 2>/dev/null); then
-            _wanted "versions-${words[3]}" \
-              expl "Available versions of ${words[3]}" \
-              compadd -- ${(f)versions}
-          fi
+          #local versions
+          #if versions=$(asdf list all "${words[3]}" 2>/dev/null); then
+          #  _wanted "versions-${words[3]}" \
+          #    expl "Available versions of ${words[3]}" \
+          #    compadd -- ${(f)versions}
+          #fi
+          _asdf__installed_versions_of_plus_system ${words[3]}
         fi
         ;;
       *)
@@ -325,12 +326,13 @@
         else
           local plugin="${words[3]}"
         fi
-        local versions
-        if versions=$(asdf list all "$plugin" 2>/dev/null); then
-          _wanted "versions-$plugin" \
-            expl "Available versions of $plugin" \
-            compadd -- ${(f)versions}
-        fi
+        #local versions
+        #if versions=$(asdf list all "$plugin" 2>/dev/null); then
+        #  _wanted "versions-$plugin" \
+        #    expl "Available versions of $plugin" \
+        #    compadd -- ${(f)versions}
+        #fi
+        _asdf__installed_versions_of_plus_system "$plugin"
         ;;
     esac
     ;;
@@ -355,5 +357,6 @@
       compset -n 4
       _normal -p "asdf-shims-${words[3]}"
     fi
+    local p_a_t_c_h_e_d
     ;;
 esac
